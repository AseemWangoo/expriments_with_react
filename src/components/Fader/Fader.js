import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import '../../App.css'

const Fader = ({ text }) => {

    const [fadeProp, setFadeProp] = useState({
        fade: 'fade-in',
    });

    useEffect(() => {
        const timeout = setInterval(() => {
            if (fadeProp.fade === 'fade-in') {
                setFadeProp({
                    fade: 'fade-out'
                })
            } else {
                setFadeProp({
                    fade: 'fade-in'
                })
            }
        }, 4000);

        return () => clearInterval(timeout)
    }, [fadeProp])

    return (
        <>
            <h1 data-testid="fader" className={fadeProp.fade}>{text}</h1>
        </>
    )
}

Fader.defaultProps = {
    text: 'Hello World!'
}

Fader.propTypes = {
    text: PropTypes.string,
}

export default Fader
