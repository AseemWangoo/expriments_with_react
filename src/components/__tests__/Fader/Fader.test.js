import { render, screen, cleanup } from "@testing-library/react";
import Fader from "../../Fader/Fader";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";

afterEach(() => {
  cleanup();
});

test("should render fader component", () => {
  render(<Fader text="test" />);

  const faderElem = screen.getByTestId("fader");

  expect(faderElem).toBeInTheDocument();
  expect(faderElem).toHaveTextContent("test");
  expect(faderElem).toHaveClass("fade-in");
});

test("fader matches snapshot", () => {
  const component = renderer.create(<Fader />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
