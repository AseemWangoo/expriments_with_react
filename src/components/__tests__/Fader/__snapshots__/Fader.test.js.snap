// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fader matches snapshot 1`] = `
<h1
  className="fade-in"
  data-testid="fader"
>
  Hello World!
</h1>
`;
